src/main/java/roomescape/service/conponent/TossPaymentClient.java

```diff
@@ -0,0 +1,44 @@
+package roomescape.service.conponent;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestClient;
+
+import roomescape.service.dto.PaymentRequestDto;
+
+@Component
+public class TossPaymentClient implements PaymentClient {
+
+    public static final String AUTHORIZATION = "Authorization";
+    private static final String AUTHORIZATION_PREFIX = "Basic ";
+    private static final String TOSS_PAYMENTS_URL = "https://api.tosspayments.com/v1/payments/confirm";
+
+    private final String authorizations;
+    private final RestClient restClient;
+
+    public TossPaymentClient(@Value("${toss-payment.test-secret-key}") String key) {
+        String tossPaymentTestKey = key + ":";
+        this.restClient = RestClient.builder()
+                .baseUrl(TOSS_PAYMENTS_URL)
+                .build();
+
+        Base64.Encoder encoder = Base64.getEncoder();
+        byte[] encodedBytes = encoder.encode(tossPaymentTestKey.getBytes(StandardCharsets.UTF_8));
+        authorizations = AUTHORIZATION_PREFIX + new String(encodedBytes);
```

Comment:
```suggestion
        this.restClient = RestClient.builder()
                .baseUrl(TOSS_PAYMENTS_URL)
                .build();

        String tossPaymentTestKey = key + ":";
        Base64.Encoder encoder = Base64.getEncoder();
        byte[] encodedBytes = encoder.encode(tossPaymentTestKey.getBytes(StandardCharsets.UTF_8));
        authorizations = AUTHORIZATION_PREFIX + new String(encodedBytes);
```
생성자에서 값 할당에서는 가능하면 사용하는 쪽에서 선언되도록 메소드라인을 적어두는 것이 가독성이 좋을 것 같아요. (생성자 뿐 아니라 메소드도 마찬가지겠네요,)

authorizations를 만드는 부분은 메소드로 분리하는건 어떨까요?

Reply:
그러네요 생성자에서 너무 많은 로직을 그대로 가지고 있어서 가독성도 좋지 않았네요..! 그러다보니 위와 같이 사용하는 메소드 라인과 변수가 너무 떨어져있었던 것 같아요!

해당 부분을 수정하고, authorizations 생성하는 로직도 따로 private 메서드로 추출하엿습니다.